### ISSUE TITLE
Timestamps in backup filenames

### ISSUE THREAD

**Issue by silic0nalph4** [jan-di/docker-database-backup#13]
Hi, thanks for the awesome tool! I've added it to a Docker Compose [cluster](https://github.com/silic0nalph4/nova-secura-10) I'm running - it's great to have automatic backups 😀

I was wondering what you thought of the following feature request:

> Add a config option that appends a timestamp to the backup filename. This would allow multiple backups to be written out from a service and remove the need for another tool to rename the files before archiving them.

I think it makes most sense for the config option to be at the database level rather than global, so it could be set for some databases and not others.

e.g.
```
# Minimal Docker Compose usage example
version: '3.8'

services:
  db-backup: # backup service
    image: ghcr.io/jan-di/database-backup
    environment:
      - TZ=Europe/Berlin
      - SCHEDULE=600
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock

  database1: # well known database image
    image: mariadb:latest
    environment:
      - MYSQL_ROOT_PASSWORD=secret-password
    labels:
      - jan-di.database-backup.enable=true
      - jan-di.database-backup.dump_timestamp=true <-- This bit is new.
```

Output;
- `/dump/database1_2021-10-30T14:35:40+00:00.sql` (ISO 8601)

Would this be something that you'd be interested in adding to this tool? If so I'm happy to have a hack at implementing it (but don't want to step on your toes!)

**Comment by jan-di** [comment]
Hi @silic0nalph4,

this is indeed a feature that is in my backlog. I want to add timestamped dumps with some kind of retention policy to cleanup the dumps. I will look into it, thanks for your feature request :-)

**Comment by jan-di** [comment]
I've not used ISO Format for the filename, because it contains colons, that are not allowed on every os as part of a filename. Currently even though the filename contains a (unique) timestamp, only one file is kept per database container. See #14 for multiple dump files. You can test it already by using the docker tag `:main`

**Comment by silic0nalph4** [comment]
That was some fast work! 

Good call on the timestamp format, I hadn't thought about path-safe characters. The format you've chosen is nicely human-readable.

#14 will be a nice further enhancement - I'm interested in a retention policy of "all" 😉 Very sensible to not change the default behaviour, though. Anybody with large backups would have been surprised!

[The build](https://github.com/jan-di/docker-database-backup/actions/runs/1403055695) for this PR died so I wasn't able to test last night. Subsequent builds succeeded so I've pulled the container image and it's running perfectly. Thanks again!

**Comment by jan-di** [comment]
This is released now in the 0.3.0 Release :-)

**Comment by silic0nalph4** [comment]
Wow, this is awesome! Thank you so much - truly great work! 🚀 So many options! And documentation!! 😍
